src/journal/journald-server.c

```diff
@@ -1188,15 +1187,16 @@ static void server_dispatch_message_real(
                 /* Split up strictly by (non-root) UID */
                 journal_uid = c->uid;
         else if (s->split_mode == SPLIT_LOGIN && c && c->uid > 0 && uid_is_valid(c->owner_uid))
-                /* Split up by login UIDs.  We do this only if the
-                 * realuid is not root, in order not to accidentally
-                 * leak privileged information to the user that is
-                 * logged by a privileged process that is part of an
-                 * unprivileged session. */
+                /* Split up by login UIDs.  We do this only if the realuid is not root, in order not to
+                 * accidentally leak privileged information to the user that is logged by a privileged
+                 * process that is part of an unprivileged session. */
                 journal_uid = c->owner_uid;
         else
                 journal_uid = 0;
 
+        if (uid_for_system_journal(journal_uid))
+                journal_uid = 0;
```

Comment:
this is really weird. why lie here?

Reply:
IMU the rest of the code is only interested in knowing whether the messages need to go to the system vs user journal. Adjusting the perms of the journal file is only needed in the latter case. For anything needing to go to the system journal, the uid of the initial sender of the message is irrelevant once we figured out the destination of the message is the system journal.

Reply:
well, sure, i understand that the precise uid is not necessary if we don't split off, but that doesn't answer the question why lying here was beneficial